automation/run/store.go

```diff
@@ -25,26 +27,26 @@ var (
 // events and updating the associated the run.State accordingly
 type Store interface {
 	// Create adds a new run State to the Store
-	Create(r *State) (*State, error)
+	Create(ctx context.Context, r *State) (*State, error)
 	// Put puts a run State with an existing run ID into the Store
-	Put(r *State) (*State, error)
+	Put(ctx context.Context, r *State) (*State, error)
 	// Get gets the associated run.State
-	Get(id string) (*State, error)
+	Get(ctx context.Context, id string) (*State, error)
 	// Count returns the number of runs for a given workflow.ID
-	Count(wid workflow.ID) (int, error)
+	Count(ctx context.Context, wid workflow.ID) (int, error)
 	// List lists all the runs associated with the workflow.ID in reverse
 	// chronological order
-	List(wid workflow.ID, lp params.List) ([]*State, error)
+	List(ctx context.Context, wid workflow.ID, lp params.List) ([]*State, error)
 	// GetLatest returns the most recent run associated with the workflow id
-	GetLatest(wid workflow.ID) (*State, error)
+	GetLatest(ctx context.Context, wid workflow.ID) (*State, error)
 	// GetStatus returns the status of the latest run based on the
 	// workflow.ID
-	GetStatus(wid workflow.ID) (Status, error)
+	GetStatus(ctx context.Context, wid workflow.ID) (Status, error)
 	// ListByStatus returns a list of run.State entries with a given status
 	// looking only at the most recent run of each Workflow
-	ListByStatus(s Status, lp params.List) ([]*State, error)
+	ListByStatus(ctx context.Context, owner profile.ID, s Status, lp params.List) ([]*State, error)
 	// Shutdown closes the store
-	Shutdown() error
+	Shutdown(ctx context.Context) error
```

Comment:
wait, do we _really_ need a context on shutdown? Strikes me as a bad idea

Reply:
Probably not, cloud should have a separate shutdown mechanic anyways (and does).